lib/webmock/http_lib_adapters/net_http.rb

```diff
@@ -119,26 +114,21 @@ def start_without_connect
           raise IOError, 'HTTP session already opened' if @started
           if block_given?
             begin
+              @socket = Net::HTTP.socket_type.new
```

Comment:
FWIW, `Net::HTTP.socket_type` is obsolete:

https://github.com/ruby/net-http/blob/196f3d70f9ce154dc63a86a2cdbf1aec4df4982e/lib/net/http.rb#L575-L577